### ISSUE TITLE
Implement Discussions for communities using Social DB

### ISSUE THREAD

**Issue by ori-near** [NEAR-DevHub/neardevhub-bos#584]
Building on https://github.com/NEAR-DevHub/neardevhub-bos/issues/540, we'd like to introduce a similar feature called Discussions. The main difference here is that any user can write a post, not just community managers of a particular community.

# Problem
Right now, it's hard for users to engage in discussions across our various communities. The existing homepage/feed is all about proposals and contributor reports, which is great, but it doesn't give people an easy way to engage.

# Solution
Create a Discussions feed on each community page, which will allow users to easily post their updates.

# User Stories
- As a user who cares about specific communities, I want to share my updates with a specific community and engage with other posts.

# Acceptance Criteria

**DevHub community page requirements:**
- [ ] There is a new Discussions tab, which is the second tab that users see when they land on a given community
- [ ] All users see a share update input text area field (e.g. What's happening)
  - [ ] The share update input text area field contains all the same elements from the near.org, including supporting markdown, image update, as well as the preview.
  - [ ] Users can type their update, preview it, upload image, and click post. Upon post, the update shows up in the discussions section with the user's profile/name.
- [ ] The discussion feed is sorted by latest, but users can also filter by recent comments
- [ ] Users can click Load More to load additional posts
- [ ] Users can interact with discussions posts (like, add comment, copy link to post, or share).
- [ ] Users can report posts (clicking the three dots). _Note: @ailisp will handle this part in a separate ticket. Please check in with him._

**near.org and near social requirements:**
- [ ] Any discussion posted on DevHub communities will also show up on near.org and near social gateways

**Backend storage requirements:**
- [ ] Use Social DB and follow near.org conventions-

# Mockups (See discussion section)
https://www.figma.com/file/6bdpB5IrHztN7E40Rjle1N/Community-Feature%3A-Announcements?type=design&node-id=12%3A6010&mode=design&t=M1jwBsC4NvLrn35p-1

# Technical Details
See technical considerations from https://github.com/NEAR-DevHub/neardevhub-bos/issues/540

**Comment by Tguntenaar** [comment]
Not sure if it is a bug but I don't see any information on this issue, like who is assigned and the description.

![Image](https://github.com/NEAR-DevHub/neardevhub-bos/assets/28901891/5762f3f7-43f5-4372-87eb-5e20e8b8224e)

**Comment by elliotBraem** [comment]
I've opened a draft PR for this; mostly complete. Will have some pending contract changes and will then prepare a preview once Announcements is merged)

**Comment by ailisp** [comment]
@elliotBraem has made annoucement to production. I think this one is quite similar in terms of UI, but one challenge is how to consider a near social post as part of the community discussion. Maybe manually add some `#community-topic` tag to the post and filter those posts when query a feed of certain community?

**Comment by frol** [comment]
> one challenge is how to consider a near social post as part of the community discussion. 

Re-post the author-posted Social Post to the community discussions account through DevHub contract.

<img width="895" alt="image" src="https://github.com/NEAR-DevHub/neardevhub-bos/assets/304265/1040ae39-f3aa-4db8-8a5f-cf01e83a4234">

And then just display the discussions account feed on the community page, it will be full of re-shared posts at the end of the day, like this: https://near.social/mob.near/widget/ProfilePage?accountId=luchito1.near
